Sorting tests

egorhm opened the pull request.

egorhm committed:
Initial commit of parametrized tests

egorhm committed:
Insertion sort stub added

egorhm committed:
[Sorting]: Fixed tests added

egorhm committed:
[Sorting]: Random generated tests added

egorhm committed:
[Sorting]: Selection and Bubble sort stubs added

egorhm committed:
[Sorting]: All stubs are added to tests

sergeyenin commented on Sorting/src/SortableTest.java:
```diff
@@ -0,0 +1,102 @@
+    @Test
+    public void sortEmptyArray() {
+        int[] array  = {};
+        assertEquals(null, sorter.sort(array));
```
why result of sorting empty array should  be null?

egorhm committed:
[Sorting]: Empty elements test fixed

tarasevich commented on Sorting/src/Sortable.java:
```diff
@@ -0,0 +1,3 @@
+public interface Sortable {
```
`Sorting` would be more accurate name (sortable is something that is sorted, sorting is an algorithm that sorts something else).

tarasevich commented on Sorting/src/SortableTest.java:
```diff
@@ -0,0 +1,102 @@
+public class SortableTest {
+    final int MAX_ARRAY_LENGTH = 10000;
+
+    private Class algorithmType = null;
```
i think it should be something like

```
private Class<? extends Sorting> algorithmType = null;
```

This way you don't need "unsafe" casts.

tarasevich commented on Sorting/src/SortableTest.java:
```diff
@@ -0,0 +1,102 @@
+    final int MAX_ARRAY_LENGTH = 10000;
+
+    private Class algorithmType = null;
+    private Sortable sorter = null;
```
`sorting`

tarasevich commented on Sorting/src/SortableTest.java:
```diff
@@ -0,0 +1,102 @@
+/**
+ * Created by Egor Hamaliy on 4/21/14.
+ */
+@RunWith(value = Parameterized.class)
```
I wonder if there such thing in RSpec or TestUnit

tarasevich commented on Sorting/src/SortableTest.java:
```diff
@@ -0,0 +1,102 @@
+            array[i] = random;
+        }
+
+        int[] sorted_array = array;
```
`Arrays.copy`

sergeyenin replied:
Alexey is right, you just copy pointer to array and then sort the array within the memory,so the assert always will be true.
You should use Arrays.copyof  or System.arraycopy.

egorhm replied:
You are right guys. At the initial implementation there needs to be clone. So, will fix to clone method.

tarasevich commented on Sorting/src/SortableTest.java:
```diff
@@ -0,0 +1,102 @@
+
+    @Test
+    public void sortOddElementsArray() {
+        int[] array  = { 100, 5, 1, 2, 9, 101, 21, 9, 2, 56, 34, 0 };
```
`assert_true 12.even?`

egorhm replied:
Will rename methods

egorhm committed:
[Sorting]: Fixes according to diff comments

sergeyenin committed:
Update SortableTest.java

to provide more information about a class implemented Sorting used within the test.

sergeyenin committed:
Added references of HeapSort and MergeSort

Alexey-Tarasevich-epam committed:
rename Sortable to Sorting
